src/Libraries/RevitNodes/Elements/Views/View.cs

```diff
@@ -99,25 +100,34 @@ public Bitmap ExportAsImage(string path)
 
             Document.ExportImage(options);
 
-            var pathName = Path.Combine(
-                            Path.GetDirectoryName(path),
-                            Path.GetFileNameWithoutExtension(path));
+            var doc = DocumentManager.Instance.CurrentDBDocument;
+            var revitName = Autodesk.Revit.DB.ImageExportOptions.GetFileName(doc, InternalView.Id);
+            var directoryName = Path.GetDirectoryName(path);
+            var actualFileName = string.Empty;
+            var destinationFileName = string.Empty;
 
-            // Revit outputs file with a bunch of crap in the file name, let's construct that
-            var actualFn = string.Format("{0} - {1} - {2}{3}", pathName, ViewTypeString(InternalView.ViewType),
-                InternalView.ViewName, extension);
+            if (directoryName != null)
+            {
+                actualFileName = Path.Combine(
+                    directoryName,
+                    Path.GetFileNameWithoutExtension(path) + revitName + extension);
+
+                destinationFileName = Path.Combine(
+                    directoryName,
+                    Path.GetFileNameWithoutExtension(path) + extension);
+            }
 
-            // and the intended destination
-            var destFn = pathName + extension;
+            if(string.IsNullOrEmpty(actualFileName) || string.IsNullOrEmpty(destinationFileName))
+                throw new ArgumentException(Properties.Resources.View_ExportAsImage_Path_Invalid, nameof(path));
 
             // rename the file
-            if (File.Exists(destFn)) File.Delete(destFn);
-            File.Move(actualFn, destFn);
+            if (File.Exists(destinationFileName)) File.Delete(destinationFileName);
```

Comment:
What happens if the user has the old image file open and there's a file lock in place? You might want to wrap this in a try/catch or check for a file lock